IntegrationTestDS3/IntegrationTestDS3Client.cs

```diff
@@ -326,6 +326,18 @@ public void TestSpecialCharacterInObjectName()
                     select f;
 
                 Assert.AreEqual(1, putfile.Count());
+
+                // Does a query param escape properly?
+                GetObjectsRequest getObjectsWithNameRequest = new GetObjectsRequest();
+                getObjectsWithNameRequest.ObjectName = fileName;
+                var getObjectsResponse = _client.GetObjects(getObjectsWithNameRequest);
+
+                var filename =
+                    from f in getObjectsResponse.Objects
+                    where f.Name == fileName
+                    select f;
+
+                Assert.AreEqual(1, filename.Count());
```

Comment:
The response will contain the filename in the xml as a string with the "+" in it, but if it wasn't converted to %2B in the outgoing request, you'd get an empty list of objects in the response (or maybe an http error?).  I also manually verified with Fiddler that it's showing up as %2B in the request.